### ISSUE TITLE
AttributeError/TypeError in Python3

### ISSUE THREAD

**Issue by jaehyunp** [cvxgrp/dccp#18]
The following test problem instance gives an attribute error in NumPy when executed with `python3`, which might have been caused by how the values are passed to the `np.max` function. The script runs fine using `python2` and terminates without an error. (Though it gives the objective value of `None`.)

Error message:
```
iteration= 1 cost value =  -inf tau =  0.005
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/numpy/core/fromnumeric.py", line 2260, in amax
    amax = a.max
AttributeError: 'list' object has no attribute 'max'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "p3test.py", line 20, in <module>
    val = prob.solve(method='dccp')[0]
  File "/usr/local/lib/python3.5/dist-packages/cvxpy-0.4.8-py3.5.egg/cvxpy/problems/problem.py", line 208, in solve
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/dccp/problem.py", line 33, in dccp
    result_temp = iter_dccp(self, max_iter, tau, mu, tau_max, solver, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/dccp/problem.py", line 386, in iter_dccp
    max_slack = np.max(max_slack)
  File "/usr/lib/python3/dist-packages/numpy/core/fromnumeric.py", line 2263, in amax
    out=out, keepdims=keepdims)
  File "/usr/lib/python3/dist-packages/numpy/core/_methods.py", line 26, in _amax
    return umr_maximum(a, axis, None, out, keepdims)
TypeError: unorderable types: NoneType() >= NoneType()
```

Code:
```python
import dccp
import cvxpy as cvx
import numpy as np
import scipy.sparse as sp

np.random.seed(1)
n = 3

x = cvx.Variable(n)
T = cvx.Variable()

obj = cvx.Minimize(T)
F1, F2 = np.random.randn(n, 2*n), np.random.randn(n, 2*n)
F1, F2 = F1.dot(F1.T), F2.dot(F2.T)
Pp, Pm = sp.csr_matrix(F1), sp.csr_matrix(F2)
cons = [cvx.quad_form(x, Pp) <= cvx.quad_form(x, Pm) + T, cvx.square(x)==1]
prob = cvx.Problem(obj, cons)
x.value = np.asmatrix(np.random.randn(n, 1))
val = prob.solve(method='dccp')[0]
print(val)
```

**Comment by xinyueshen** [comment]
The error is caused by comparison between NoneType, which is allowed in Python 2 but not in Python 3.  The values of slack variables become NoneType when the subproblem that we are trying to solve is infeasible or unbounded, so I fixed it by trying to get the max slack value only when a solution is found for the subproblem.

For this test problem, it can find a feasible point if we set tau = 1, i.e.,
``val = prob.solve(method='dccp', tau = 1)[0]``
Any value for tau less than 1 will lead the subproblem to be unbounded.
